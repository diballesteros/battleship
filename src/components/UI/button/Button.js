import React from 'react';
import './Button.css';

const Button = (props) => (
    <button onClick={props.clicked} disabled={props.disabled}>
        {props.children}
    </button>
);

export default Button;
