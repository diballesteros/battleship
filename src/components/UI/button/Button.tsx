import React from 'react';
import './Button.css';

interface ButtonProps {
  children: string;
  clicked: () => void;
  disabled: boolean;
}

const Button: React.FC<ButtonProps> = ({ children, clicked, disabled }) => (
  <button onClick={clicked} disabled={disabled} type="button">
    {children}
  </button>
);

export default Button;
