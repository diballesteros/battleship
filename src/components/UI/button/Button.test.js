import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Button from './Button';

describe('Button component', () => {
    test('Button initalized', () => {
        const { asFragment } = render(<Button disabled={false}>Start Game</Button>);
        expect(asFragment()).toMatchSnapshot();
    });

    test('Button disabled', () => {
        const { asFragment } = render(<Button disabled={true}>Start Game</Button>);
        expect(asFragment()).toMatchSnapshot();
    });

    test('Button click', () => {
        const mockFn = jest.fn();
        const { container } = render(<Button disabled={false} clicked={() => mockFn()}>Start Game</Button>);
        const component = container.querySelector('button');
        fireEvent.click(component);
        expect(mockFn).toHaveBeenCalled();
    });
});
