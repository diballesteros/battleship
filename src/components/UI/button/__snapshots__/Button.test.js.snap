// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component Button disabled 1`] = `
<DocumentFragment>
  <button
    disabled=""
  >
    Start Game
  </button>
</DocumentFragment>
`;

exports[`Button component Button initalized 1`] = `
<DocumentFragment>
  <button>
    Start Game
  </button>
</DocumentFragment>
`;
